src/AfterLogin/ReportPage.js

```diff
@@ -0,0 +1,143 @@
+import { Backdrop, Button, CircularProgress } from "@mui/material";
+import { useEffect, useMemo, useState } from "react";
+import { useOutletContext, Link, useNavigate } from "react-router-dom";
+import Divider from "@mui/material/Divider";
+import "./ReportPage.css";
+import { CartesianGrid, Line, LineChart, XAxis, YAxis } from "recharts";
+import ErrorPage from "../ErrorPage";
+import DBHandler from "../Controller/DBHandler";
+//Take the user data from App.js state
+
+export default function ReportPage() {
+  const [user] = useOutletContext();
+  const [userInfo, setUserInfo] = useState(null);
+  const [userWords, setUserWords] = useState(0);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [goHome, setGoHome] = useState(false);
+  const navi = useNavigate();
+  const dbHandler = useMemo(
+    () => new DBHandler(user.uid, setErrorMessage, setGoHome),
+    [user.uid, setErrorMessage, setGoHome]
+  );
+
+  const handleErrorMessage = () => {
+    setErrorMessage("");
+    if (goHome) {
+      navi("/");
+    }
+  };
+
+  useEffect(() => {
+    const getUserAndDeckInfo = async () => {
+      try {
+        const { userInfo, userDecks } = await dbHandler.getUserAndDecksInfo(
+          true
+        );
+        const words = new Set();
+        for (const deck of userDecks) {
+          for (const cardID of deck.deckCards) {
+            words.add(cardID);
+          }
+        }
+        setUserInfo(userInfo);
+        setUserWords(words.size);
```

Comment:
You could probably define a fetcher function to feed into a custom hook, which will then return variables as necessary